framework/test/driver/ResourceCompilerTest.cpp

```diff
@@ -422,28 +536,569 @@ static void testrcFailureModes(const std::string& rcbinpath, const std::string&
+{
+  std::string dupKeys(R"(
+		{
+		 "bundle.name": "foo",
+		 "bundle.version" : "1.0.2",
+		 "bundle.description" : "This bundle is broken!",
+		 "authors" : ["John Doe", "Douglas Reynolds", "Daniel Cannady"],
+		 "bundle.name" : "foobar",
+		 "rating" : 5
+		}
+    )");
+
+  US_TEST_NO_EXCEPTION_REQUIRED(createManifestFile(tempdir, dupKeys));
+
+  std::ostringstream cmd;
+  cmd << rcbinpath;
+  cmd << " --bundle-name " << "mybundle";
+  cmd << " --out-file " << tempdir << "duplicate_keys.zip";
+  cmd << " --manifest-add " << tempdir << "manifest.json";
+  US_TEST_CONDITION(BUNDLE_MANIFEST_VALIDATION_ERROR_CODE == runExecutable(cmd.str()), "Fail to embed manifest containing duplicate JSON key names.");
+
+  ZipFile duplicate_keys_zip(tempdir + "duplicate_keys.zip");
+  US_TEST_CONDITION(0 == duplicate_keys_zip.size(), "Test that the invalid manifest.json file was not added");
+
+  std::string duplicate_keys_bundle_file("duplicate_keys.bundle");
+  std::ostringstream cmd2;
+  cmd2 << rcbinpath;
+  cmd2 << " --bundle-name " << "mybundle";
+  cmd2 << " --bundle-file " << tempdir << duplicate_keys_bundle_file;
+  cmd2 << " --manifest-add " << tempdir << "manifest.json";
+  US_TEST_CONDITION(BUNDLE_MANIFEST_VALIDATION_ERROR_CODE == runExecutable(cmd2.str()), "Fail to embed manifest containing duplicate JSON key names.");
+
+  US_TEST_CONDITION(false == containsBundleZipFile(tempdir + duplicate_keys_bundle_file), "Test that the invalid manifest.json file was not added");
+
+  std::ostringstream cmd3;
+  cmd3 << rcbinpath;
+  cmd3 << " --bundle-name mybundle ";
+  cmd3 << " --out-file duplicate_keys_res_add.zip ";
+  cmd3 << " --res-add manifest.json";
+
+  auto origdir = util::GetCurrentWorkingDirectory();
+  testing::ChangeDirectory(tempdir);
+  US_TEST_CONDITION(BUNDLE_MANIFEST_VALIDATION_ERROR_CODE == runExecutable(cmd3.str()), "Fail to embed manifest containing duplicate JSON key names.");
+  testing::ChangeDirectory(origdir);
+
+  ZipFile duplicate_keys_res_add_zip(tempdir + "duplicate_keys_res_add.zip");
+  US_TEST_CONDITION(0 == duplicate_keys_res_add_zip.size(), "Test that the invalid manifest.json file was not added");
+
+  std::string duplicate_keys_res_add_bundle_file("duplicate_keys_res_add.bundle");
+  std::ostringstream cmd4;
+  cmd4 << rcbinpath;
+  cmd4 << " --bundle-name mybundle ";
+  cmd4 << " --bundle-file " << duplicate_keys_res_add_bundle_file;
+  cmd4 << " --res-add manifest.json";
+
+  testing::ChangeDirectory(tempdir);
+  US_TEST_CONDITION(BUNDLE_MANIFEST_VALIDATION_ERROR_CODE == runExecutable(cmd4.str()), "Fail to embed manifest containing duplicate JSON key names.");
+  testing::ChangeDirectory(origdir);
+  US_TEST_CONDITION(false == containsBundleZipFile(tempdir + duplicate_keys_res_add_bundle_file), "Test that the invalid manifest.json file was not added");
+}
+
+// A zip file containing only a manifest.
+// The zip file is deleted when this object is destroyed.
```

Comment:
`// The zip file is deleted when this object is destroyed.`. the deletion code is missing from the destructor?

Reply:
I forgot to remove this comment. There is no need to delete the zip file now that the temporary directory and everything in it is completely cleaned up after running the tests.